### PR TITLE
doc: document for oma-console

### PR DESCRIPTION
**Pull request by wxiwnd**

### CODE REVIEW

File: oma-console/src/pager.rs [hunk 1]
```diff
@@ -107,6 +108,28 @@ enum PagerInner {
     Finished(Vec<String>),
 }
 
+///
+/// `OmaPager` is a structure that represents a pager displaying content in a terminal UI.
+///
+/// # Fields
+///
+/// * `inner` - Represents the internal state of the pager, which can be either `Working` or `Finished`.
```
Comment (eatradish) [inline oma-console/src/pager.rs:116]: Write the comment on top of the field, for example:

```rust
pub struct OmaPager<'a> {
    /// Represents the internal state of the pager, which can be either `Working` or `Finished`.
    inner: PagerInner,
    /// The state of the vertical scrollbar.
    vertical_scroll_state: ScrollbarState,
    /// The state of the horizontal scrollbar.
    horizontal_scroll_state: ScrollbarState,
    ...
```

Only helps the IDE read field comments.

File: oma-console/src/print.rs [hunk 2]
```diff
@@ -116,7 +134,7 @@ fn term_color<D>(input: D, color: Action) -> StyledObject<D> {
         Action::PendingBg => style(input).bg(Color::Blue).bold(),
     }
 }
-
+/// OmaLayer
```
Comment (eatradish) [inline oma-console/src/print.rs:137]: `OmaLayer` and `OmaRecorder` are used to output oma-style logs to `tracing`, such as in `src/main.rs:206`:

```rust
    if !debug {
        let no_i18n_embd_info: EnvFilter = "i18n_embed=off,info".parse().unwrap();

        tracing_subscriber::registry()
            .with(
                OmaLayer
                    .with_filter(no_i18n_embd_info)
                    .and_then(LevelFilter::INFO),
            )
            .init();
    } 
```

**Review by eatradish — CHANGES_REQUESTED** [review]

File: oma-console/src/print.rs [hunk 3]
```diff
@@ -56,7 +56,14 @@ impl Action {
         }
     }
 }
-
+/// OmaColorFormat
```
Comment (eatradish) [inline oma-console/src/print.rs:59]: above

File: oma-console/src/pager.rs (hunk 1 above)
Comment (wxiwnd) [reply oma-console/src/pager.rs]: Fixed in https://github.com/AOSC-Dev/oma/pull/228/commits/b5856e7e5ff0dc37abdf09b60dae02a101049901

File: oma-console/src/print.rs (hunk 2 above)
Comment (wxiwnd) [reply oma-console/src/print.rs:137]: Fixed this in https://github.com/AOSC-Dev/oma/pull/228/commits/2d03568ca097c37f4812813b1688a6d68434c17c

**Review by eatradish — APPROVED** [review]

File: oma-console/src/print.rs (hunk 3 above)
Comment (wxiwnd) [reply oma-console/src/print.rs:59]: https://github.com/AOSC-Dev/oma/pull/228/commits/9f4f5f4c4222db539a3f32aafacd10ad2848c82b

**Review by eatradish — APPROVED** [review]

File: oma-console/src/pager.rs [hunk 4]
```diff
@@ -107,21 +108,37 @@ enum PagerInner {
     Finished(Vec<String>),
 }
 
+/// `OmaPager` is a structure that represents a pager displaying content in a terminal UI.
```
[the remark this answers is not in the document]
Comment (MingcongBai) [reply oma-console/src/pager.rs:111]: text-based content

File: oma-console/src/print.rs [hunk 5]
```diff
@@ -84,7 +88,18 @@ impl OmaColorFormat {
             },
         }
     }
-
+    /// Convert input into StyledObject
+    ///
+    /// This function applies a color to the given input string based on the specified action and the current theme settings.
```
Comment (MingcongBai) [inline oma-console/src/print.rs:93]: a color scheme

Comment (MingcongBai) [reply oma-console/src/print.rs:93]: and the current terminal color schemes

File: oma-console/src/print.rs [hunk 6]
```diff
@@ -84,7 +88,18 @@ impl OmaColorFormat {
...
+    ///
+    /// # Arguments
+    ///
+    /// * `input` - The input data to be styled.
```
Comment (MingcongBai) [inline oma-console/src/print.rs:97]: The input data to be themed

File: oma-console/src/print.rs [hunk 7]
```diff
@@ -116,7 +131,22 @@ fn term_color<D>(input: D, color: Action) -> StyledObject<D> {
         Action::PendingBg => style(input).bg(Color::Blue).bold(),
     }
 }
-
+/// OmaLayer
+/// `OmaLayer` is used for output oma-style logs to `tracing`
```
Comment (MingcongBai) [inline oma-console/src/print.rs:135]: is used to

File: oma-console/src/print.rs [hunk 8]
```diff
@@ -148,7 +178,19 @@ where
         }
     }
 }
-
+/// OmaRecorder
+/// `OmaRecorder` is used for record oma-style logs.
```
Comment (MingcongBai) [inline oma-console/src/print.rs:182]: is used to

File: oma-console/src/print.rs (hunk 7 above)
Comment (wxiwnd) [reply oma-console/src/print.rs:135]: ”is used to“ means someone did it regularly in the past, and "is used for" indicates the purpose. There is no grammar issues.

File: oma-console/src/print.rs (hunk 8 above)
Comment (wxiwnd) [reply oma-console/src/print.rs:182]: same as above

**Pull request merged**